src/app/components/QrcodeAdornment/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { QrCodeIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
+import { useNavigate } from "react-router-dom";
+
+type Props = {
+  route: string;
+};
+
+export default function QrcodeAdornment({ route }: Props) {
+  const navigate = useNavigate();
+  return (
+    <button
+      aria-label="Scan QR"
+      type="button"
+      className="flex justify-center items-center w-10 h-8"
+      onClick={() => {
+        navigate("/scanQRCode", { state: { route: route }, replace: true });
```

Comment:
This should not be `replace: true`, right? When you exit instead of going back to /send or /lnurlRedeem it will go back two pages

Reply:
this is needed, cause if we go back we will have two routes in navigation stack for same page. instead, i changed code for exist button to navigate with replace: true. and things work fine for both back button and exit button

Reply:
Reason to do this:

* we go to lnulr redeem screen - first entry in stack
* we go to scan qr code screen via qr adornment component - second entry in stack - it replaces previous lnurl redeem screen present in stack
* we scan qr and go back to lnurl redeem screen. we remove scanQrcode screen entry
* in the end we will have single lnurl-redeem screen entry
* now if we press back button it will go directly to receive

Reply:
sounds good, thank you